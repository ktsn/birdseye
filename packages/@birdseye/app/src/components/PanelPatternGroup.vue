<template>
  <div class="panel-pattern-group">
    <strong class="title">
      {{ title }}
    </strong>

    <ul
      v-if="data.length > 0"
      class="list"
    >
      <li
        v-for="item in data"
        :key="item.name"
        class="item"
      >
        <strong class="name">
          {{ item.name }}
        </strong>

        <span class="value">
          {{ item.value }}
        </span>
      </li>
    </ul>

    <p
      v-else
      class="no-data"
    >
      No Data
    </p>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { PatternData } from './PanelPattern.vue'

export default Vue.extend({
  name: 'PanelPatternGroup',

  props: {
    title: {
      type: String,
      required: true
    },

    data: {
      type: Array as () => PatternData[],
      required: true
    }
  }
})
</script>

<style scoped>
.title {
  display: block;
  font-weight: normal;
  font-size: var(--font-size-large);
}

.item {
  margin-top: 12px;
}

.name,
.value {
  display: inline-block;
}

.name {
  margin-right: 0.5em;
  font-weight: bold;
}

.no-data {
  margin-top: 12px;
  color: #888;
}
</style>
