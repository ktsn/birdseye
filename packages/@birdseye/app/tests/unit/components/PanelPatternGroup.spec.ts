import { shallowMount } from '@vue/test-utils'
import PanelPatternGroup from '@/components/PanelPatternGroup.vue'

describe('PanelPatternGroup', () => {
  it('renders title and data', () => {
    const data = [
      {
        type: ['string'],
        name: 'foo',
        value: 'string value'
      },
      {
        type: ['number'],
        name: 'bar',
        value: 123
      }
    ]

    const wrapper = shallowMount(PanelPatternGroup, {
      propsData: {
        title: 'title string',
        data
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders no data when data is empty', () => {
    const wrapper = shallowMount(PanelPatternGroup, {
      propsData: {
        title: 'no data',
        data: []
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
