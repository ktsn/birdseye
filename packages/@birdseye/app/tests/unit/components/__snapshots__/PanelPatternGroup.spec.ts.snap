// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelPatternGroup renders no data when data is empty 1`] = `
<div class="panel-pattern-group"><strong class="title">
    no data
  </strong>
  <p class="no-data">
    No Data
  </p>
</div>
`;

exports[`PanelPatternGroup renders title and data 1`] = `
<div class="panel-pattern-group"><strong class="title">
    title string
  </strong>
  <ul class="list">
    <li class="item"><strong class="name">
        foo
      </strong> <span class="value">
        string value
      </span></li>
    <li class="item"><strong class="name">
        bar
      </strong> <span class="value">
        123
      </span></li>
  </ul>
</div>
`;
